### DISCUSSION TITLE
PAC CLI Create Connectors

### DISCUSSION THREAD

**Discussion by StylezDevops** [microsoft/powerplatform-build-tools#549] [Ideas]
@kkanakas

I understand the reasons WHY it is not possible currently but would like to see the ability to create connectors (standard/premium) programmatically.  An official task would be great but I would also settle for the functionality to be built into pac cli.

Whilst pondering this I also thought it worth mentioning that if this does get actioned at some point - it would be nice if you could specify the connector required by name - so I don't have to go off and find a GUID every time i want to use it.

`pac connector create "Microsoft Dataverse"`

or something similar?

**Comment by petrochuk** [comment]
`pac connector create` needs either `--api-definition-file` or `--settings-file`. Can you clarify when do you need GUID?

**Comment by StylezDevops** [comment]
aah sorry i wasn't clear... i know it uses definition or settings file.. what i meant was.. .if and when we get the ability to create STANDARD/PREMIUM connectors... it would be nice if we could do something like:

pac connector create "Connector Name"

and avoid the usage of any GUID - i wouldn't like having to do:

pac connector create "RANDOM-GUID-HERE"

This was not about any existing functionality. Rather a wishlist of how I would like the creation of a standard/premium connector using pac cli.
Regards
Dan

**Comment by Laskewitz** [comment]
@StylezDevops Are you referring to custom connectors or connections? It sounds more like the latter.

`pac connector` commands are for creating, downloading, updating, and deleting custom connectors. All custom connectors are premium by default.

**Comment by StylezDevops** [comment]
No i mean normal connectors. Here's what I'd like to see once the connectors are moved from the "RP" layer

The ability to create Connectors such as those listed here:
https://learn.microsoft.com/en-us/connectors/connector-reference/connector-reference-production-connectors

Currently there is not a way to create a connector without using the "UI" - this means there is ALWAYS a requirement to setup connectors in a target environment as part of any enterprise deployment. The ability to CREATE these connectors would be huge. We already have the ability to create an environment from the command line. With the ability to create connectors we move one step closer to true ALM. I can then automate the creation of the environment, create connectors and prep an environment for solution deployment all within the context of a pipeline. This is my desired end goal.
